Userland/Applications/TaskbarSettings/TaskbarSettingsWidget.cpp

```diff
@@ -0,0 +1,119 @@
+    auto& taskbar_left_radio = *find_descendant_of_type_named<GUI::RadioButton>("taskbar_left_radio");
+    auto& taskbar_top_radio = *find_descendant_of_type_named<GUI::RadioButton>("taskbar_top_radio");
+    auto& taskbar_right_radio = *find_descendant_of_type_named<GUI::RadioButton>("taskbar_right_radio");
+
+    m_taskbar_location = parse_location(Config::read_string("Taskbar", "Appearance", "Location"));
+    m_original_taskbar_location = m_taskbar_location;
+
+    switch (m_taskbar_location) {
+    case Gfx::Alignment::Bottom:
+        taskbar_bottom_radio.set_checked(true);
+        break;
+    case Gfx::Alignment::Left:
+        taskbar_left_radio.set_checked(true);
+        break;
+    case Gfx::Alignment::Top:
+        taskbar_top_radio.set_checked(true);
+        break;
+    case Gfx::Alignment::Right:
+        taskbar_right_radio.set_checked(true);
+        break;
+    case Gfx::Alignment::Center:
+        dbgln("Taskbar location can't be center!");
+        VERIFY_NOT_REACHED();
+    }
+
+    taskbar_bottom_radio.on_checked = [this](bool) {
+        m_taskbar_location = Gfx::Alignment::Bottom;
+        Config::write_string("Taskbar", "Appearance", "Location", stringify_location(m_taskbar_location));
+    };
+    taskbar_left_radio.on_checked = [this](bool) {
+        m_taskbar_location = Gfx::Alignment::Left;
+        Config::write_string("Taskbar", "Appearance", "Location", stringify_location(m_taskbar_location));
+    };
+    taskbar_top_radio.on_checked = [this](bool) {
+        m_taskbar_location = Gfx::Alignment::Top;
+        Config::write_string("Taskbar", "Appearance", "Location", stringify_location(m_taskbar_location));
+    };
+    taskbar_right_radio.on_checked = [this](bool) {
+        m_taskbar_location = Gfx::Alignment::Right;
+        Config::write_string("Taskbar", "Appearance", "Location", stringify_location(m_taskbar_location));
+    };
+}
+
+Gfx::Alignment TaskbarSettingsMainWidget::parse_location(StringView location)
+{
+    if (location == "Bottom")
+        return Gfx::Alignment::Bottom;
+    if (location == "Left")
+        return Gfx::Alignment::Left;
+    if (location == "Top")
+        return Gfx::Alignment::Top;
+    if (location == "Right")
+        return Gfx::Alignment::Right;
+    VERIFY_NOT_REACHED();
+}
+
+String TaskbarSettingsMainWidget::stringify_location(Gfx::Alignment location)
+{
+    if (location == Gfx::Alignment::Bottom)
+        return "Bottom";
+    if (location == Gfx::Alignment::Left)
+        return "Left";
+    if (location == Gfx::Alignment::Top)
+        return "Top";
+    if (location == Gfx::Alignment::Right)
+        return "Right";
+    VERIFY_NOT_REACHED();
+}
+
+void TaskbarSettingsMainWidget::apply_settings()
+{
+    m_original_taskbar_location = m_taskbar_location;
+    write_back_settings();
+}
+void TaskbarSettingsMainWidget::write_back_settings() const
+{
+    Config::write_string("Taskbar", "Appearance", "Location", stringify_location(m_original_taskbar_location));
```

Comment:
`stringify` isn't really a word we use in Serenity code. Perhaps `location_to_string` would be more appropriate here.

Reply:
Stringify was used in the terminal settings application so I adopted it from there. Will update it to a more appropriate name.